addons/l10n_in/views/account_move_line_views.xml

```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <record id="view_move_line_tree_hsn_l10n_in" model="ir.ui.view">
+        <field name="name">account.move.line.tree.l10n_in</field>
+        <field name="model">account.move.line</field>
+        <field name="arch" type="xml">
+            <tree string="Products" editable="top" create="0">
+                <field name="product_id" domain="context.get('default_move_type') in ('out_invoice', 'out_refund', 'out_receipt') and [('sale_ok', '=', True)] or [('purchase_ok', '=', True)]"/>
```

Comment:
The product_id could stay readonly and then we do not need the complex domain.  Same for the name.  The goal of the view is to change the HSN.